### ISSUE TITLE
FlyoutItems duplicated on setting FlyoutItemIsVisible

### ISSUE THREAD

**Issue by ehoward0018** [dotnet/maui#11460]
### Description

When the FlyoutItemIsVisible property is set to false at runtime, the FlyoutItem itself disappears correctly, but the remaining items are duplicated. 

### Steps to Reproduce

1. Launch app
2. Go to all three flyouts
3. Go back to first flyout
4. Press button
5. Hover below flyouts and you can see the old ones are still there, now invisible.

### Link to public reproduction project repository

https://github.com/ehoward0018/dnm-flyout-bug

### Version with bug

7.0 (current)

### Last version that worked well

6.0.424

### Affected platforms

iOS, Android, Windows, macOS

### Affected platform versions

Windows 10.0 SDK

### Did you find any workaround?

I can't see how to resolve this. The app is rebuilding the flyouts, and not throwing away the old ones.

**Comment by ehoward0018** [comment]
I have developed a bit of a workaround for anyone who has been following this issue.

If you want to have different sets of flyouts for different types of users at runtime, instead of having a single AppShell, create one for each type of user.

When you want to change to a certain users set of flyouts simply do: 

`App.Current.MainPage = new <InsertNewAppShellClass>();`

As far as I can tell this works without any problems.

So in summary, never change the visibility of any of the flyouts, just have different app shells with different flyouts inside them.

**Comment by hellqvist** [comment]
I've the same problem but it doesn't matter weather I set the FlyoutItemIsVisible property or not. 

I have dynamically added items and as soon as a new item is added in code behind a new set of flyout items is rendered without the old set being removed from the flyout. I've checked and the items in the actual list isn't being duplicated, only what's rendered. 

It doesn't seem to matter if the "original" items is added in Xaml or in code behind.

**Comment by homeyf** [comment]
Verified this issue with Visual Studio Enterprise 17.7.0 Preview 1.0. Can repro on Windows platform with sample project.
https://github.com/ehoward0018/dnm-flyout-bug
![demo2](https://github.com/dotnet/maui/assets/82450723/829ce290-0dac-4471-8851-4f8da60aeaae)
